python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -222,54 +225,62 @@ def _update(self):
             nodes_allowed_to_terminate = self._get_nodes_allowed_to_terminate(
                 sorted_node_ids)
 
+        nodes_to_keep = []
+
+        def keep_node(node_id) -> None:
+            # Update per-type counts.
+            tags = self.provider.node_tags(node_id)
+            if TAG_RAY_USER_NODE_TYPE in tags:
+                node_type = tags[TAG_RAY_USER_NODE_TYPE]
+                node_type_counts[node_type] += 1
+            nodes_to_keep.append(node_id)
+
+        def terminate_node(node_id, kind, why=None) -> None:
+            why = why or kind
```

Comment:
Why don't we remove this entirely and make this just take in "terminate_reason" argument. I see that you are using `terminate_node(node_id, "unneeded", "max workers")` which we could replace with `terminate_node(node_id, "max workers")`

Reply:
sounds good